OpenRA.Game/Input/NamedHotkey.cs

```diff
@@ -0,0 +1,40 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+
+namespace OpenRA
+{
+	/// <summary>
+	/// A reference to either a named hotkey (defined in the game settings) or a statically assigned hotkey
+	/// </summary>
+	public class NamedHotkey
+	{
+		readonly Func<Hotkey> getValue = () => Hotkey.Invalid;
+
+		public NamedHotkey() { }
+		public NamedHotkey(string name, KeySettings settings)
+		{
+			// Try parsing the value as a reference to a named hotkey
+			getValue = settings.GetHotkeyReference(name);
+
+			// Try parsing the value as a normal (static) hotkey
+			Hotkey staticKey;
+			if (getValue == null && Hotkey.TryParse(name, out staticKey))
+				getValue = () => staticKey;
+		}
```

Comment:
getValue returns either Invalid or null. In case of a mistype hotkey reference, OpenRA crashes in another place in various times/situations.
If you want to crash it in such case, it could crash here or at least log invalid hotkey name.